FirebaseDatabaseUITests/FirebaseArrayTest.m

```diff
@@ -378,4 +378,36 @@ - (void)testFirebaseArrayCanMoveElement {
   XCTAssert(expectedParametersWereCorrect, @"unexpected parameter in delegate callback");
 }
 
+- (void)testFirebaseArrayCanMoveElementToStart {
+    [self.observable populateWithCount:10];
+    self.snap.key = @"8";
+    
+    // Test delegate
+    __block BOOL delegateWasCalled = NO;
+    __block BOOL expectedParametersWereCorrect = NO;
+    self.arrayDelegate.didMoveObject = ^(FirebaseArray *array, id object, NSUInteger from, NSUInteger to) {
+        // Xcode complains about retain cycles if an XCTAssert is placed in here.
+        delegateWasCalled = YES;
+        expectedParametersWereCorrect = (array == self.firebaseArray &&
+                                         object == self.snap &&
+                                         from == 8 && to == 0);
+    };
+    
+    // Move 8 to the start
+    [self.observable sendEvent:FIRDataEventTypeChildMoved withObject:self.snap previousKey:@"" error:nil];
```

Comment:
Can you add a case like this where previous key is `nil` as well? Other than that this change looks good.